src/table/system.rs

```diff
@@ -68,6 +77,17 @@ impl<View: SystemTableView> SystemTable<View> {
         self.table.header.revision
     }
 
+    /// Checks whether this table is Boot or Runtime
+    pub fn state(&self) -> SystemTableViewStatus {
+        if TypeId::of::<View>() == TypeId::of::<Boot>() {
```

Comment:
nit: wouldn't a `match` look nicer here?

PS: I never used the `TypeId` magic of Rust but this looks like a cool usage!

Reply:
See https://github.com/rust-osdev/uefi-rs/pull/491#issuecomment-1229244401 though, I don't think using static types will work here for this use case.